dirsrvtests/tests/suites/plugins/attruniq_test.py

```diff
@@ -324,4 +324,65 @@ def test_exclude_subtrees(topology_st):
         cont2.delete()
         cont3.delete()
         attruniq.disable()
-        attruniq.delete()
\ No newline at end of file
+        attruniq.delete()
+
+
+def test_matchingrule_attr(topology_st):
+    """ Test list extension MR attribute. Check for "cn" using CES (versus it
+    being defined as CIS)
+
+    :id: 5cde4342-6fa3-4225-b23d-0af918981075
+    :setup: Standalone instance
+    :steps:
+        1. Setup and enable attribute uniqueness plugin to use CN attribute
+           with a matching rule of CaseExactMatch.
+        2. Add user with CN value is lowercase
+        3. Add second user with same lowercase CN which should be rejected
+        4. Add second user with same CN value but with mixed case
+        4. Modify second user replacing CN value to lc which should be rejected
+
+    :expectedresults:
+        1. Success
+        2. Success
+        3. Success
+        4. Success
+        5. Success
+    """
+
+    inst = topology_st.standalone
+
+    attruniq = AttributeUniquenessPlugin(inst,
+                                         dn="cn=attribute uniqueness,cn=plugins,cn=config")
+    attruniq.add_unique_attribute('cn:CaseExactMatch:')
```

Comment:
Should we also add `(cn:CaseExactMatch:=Value)` test - hence, to add the `value` part?

Reply:
> Should we also add `(cn:CaseExactMatch:=Value)` test - hence, to add the `value` part?

I don't think that will work as this value is supposed to be the "attribute".  The plugin adds the "value" through its config attribute: 

    uniqueness-attribute-name: krbCanonicalName:CaseIgnoreMatch:
 
This is converted to a search filter in preop_add()/preop_modify():   

    krbCanonicalName:CaseIgnoreMatch:=<value from client's add/mod>

Reply:
I don't think we document anyway about the matching rule filter option.  Attribute uniqueness plugin did not support it until Alexanders PR that was pushed a few days ago.